attack_mapper.py

```diff
@@ -0,0 +1,466 @@
+            }
+        elif typ == "relationship":
+            src = o.get("source_ref")
+            dst = o.get("target_ref")
+            if src and dst:
+                relationships_by_src.setdefault(src, []).append(o)
+                relationships_by_dst.setdefault(dst, []).append(o)
+        elif typ in ("x-mitre-data-component", "x_mitre_data_component"):
+            data_components_by_id[o.get("id")] = o
+        elif "analytic" in typ:  # x-mitre-analytic
+            analytics_by_id[o.get("id")] = o
+
+    # Detect v18 presence: any detection-strategy SDOs or x-mitre-analytic
+    has_v18_strategy = any((o.get("type") or "").lower() in ("x-mitre-detection-strategy", "x_mitre_detection_strategy") for o in objs)
+    has_v18_analytic = bool(analytics_by_id)
+    has_v18 = has_v18_strategy or has_v18_analytic
+
+    # Build v18 indices if present
+    if has_v18:
+        # a) First, collect strategies w/ friendly DET id
+        strategies_by_id: Dict[str, Dict[str, Any]] = {}
+        for o in objs:
+            typ = (o.get("type") or "").lower()
+            if typ in ("x-mitre-detection-strategy", "x_mitre_detection_strategy"):
+                det_id = None
+                for er in (o.get("external_references") or []):
+                    ext = er.get("external_id", "") or ""
+                    m = re.match(r'^DET-?(\d{4})$', str(ext).strip(), flags=re.IGNORECASE)
+                    if m:
+                        det_id = f"DET{m.group(1)}"  # normalize to DET0001
+                        break
+
+                strategies_by_id[o.get("id")] = {
+                    "obj": o,
+                    "det_id": det_id,
+                }
+
+        # b) Map strategies to techniques via explicit detects relationships (if present)
+        for s_id, pack in strategies_by_id.items():
+            s_obj = pack["obj"]
+            det_id = pack["det_id"]
+            s_row = {
+                "id": s_id,
+                "name": s_obj.get("name", ""),
+                "summary": s_obj.get("description", ""),
+                "external_references": s_obj.get("external_references", []),
+                "det_id": det_id,
+            }
+            for rel in relationships_by_src.get(s_id, []):
+                if (rel.get("relationship_type") or "").lower() != "detects":
+                    continue
+                tgt = rel.get("target_ref")
+                t_tid = _tid_from_attack_pattern_id(tgt, techniques_by_id)
+                if not t_tid:
+                    continue
+                p_tid = _parent_tid(t_tid)
+                strategies_by_tid.setdefault(t_tid, []).append(s_row)
+                if t_tid != p_tid:
+                    strategies_by_tid.setdefault(p_tid, []).append(s_row)
+
+        # c) Regardless of relationships, also bind strategies/analytics to techniques by
+        #    extracting TIDs from each referenced analytic's id/name/refs.
+        for s_id, pack in strategies_by_id.items():
```

Comment:
The ATT&CK page maps an analytic to a specific detection, which in turn is mapped to a specific (sub)technique. If we want to map an analytic to a (sub)technique ID, we can do either of the following:
1. for `x-mitre-analytic` type bundle objects, extract the DETXXXX ID from the `external_references.url` field, and then look up the corresponding TID for that DET ID. Save the analytic object under that TID and platform
2. store analytic objects by ref ID like we do in the initial first pass. Then as we iterate through the detection strategy objects in above loop (part B) where we map strategies to TIDs, we can map the corresponding analytic(s) to the same TIDs